internal/command/action_v2_execution_model.go

```diff
@@ -64,3 +62,52 @@ func ExecutionAggregateFromWriteModel(wm *eventstore.WriteModel) *eventstore.Agg
 		Version:       execution.AggregateVersion,
 	}
 }
+
+type ExecutionsExistWriteModel struct {
+	eventstore.WriteModel
+
+	ids         []string
+	existingIDs []string
+}
+
+func (e *ExecutionsExistWriteModel) AllExists() bool {
+	return len(e.ids) == len(e.existingIDs)
+}
+
+func NewExecutionsExistWriteModel(ids []string, resourceOwner string) *ExecutionsExistWriteModel {
+	return &ExecutionsExistWriteModel{
+		WriteModel: eventstore.WriteModel{
+			ResourceOwner: resourceOwner,
+			InstanceID:    resourceOwner,
+		},
+		ids: ids,
+	}
+}
+
+func (wm *ExecutionsExistWriteModel) Reduce() error {
+	for _, event := range wm.Events {
+		switch e := event.(type) {
+		case *execution.SetEvent:
+			if !slices.Contains(wm.existingIDs, e.ID) {
+				wm.existingIDs = append(wm.existingIDs, e.ID)
+			}
+		case *execution.RemovedEvent:
+			i := slices.Index(wm.existingIDs, e.ID)
+			if i >= 0 {
+				wm.existingIDs = slices.Delete(wm.existingIDs, i, i)
+			}
```

Comment:
Perhaps add an unit test case for the removed event?

Reply:
Luckily you mentioned tests, added them and found that exactly this line had a problem it it with the second given number.